crates/amaru/src/ledger/rewards.rs

```diff
@@ -727,17 +730,18 @@ impl RewardsSummary {
             },
         );
 
-        accounts.insert(
-            reward_account_to_stake_credential(&pool.parameters.reward_account).unwrap_or_else(
-                || {
-                    panic!(
-                        "unexpected malformed reward account: {:?}",
-                        &pool.parameters.reward_account
-                    )
-                },
-            ),
-            rewards_leader,
-        );
+        let credential = reward_account_to_stake_credential(&pool.parameters.reward_account)
+            .unwrap_or_else(|| {
+                panic!(
+                    "unexpected malformed reward account: {:?}",
+                    &pool.parameters.reward_account
+                )
+            });
+
+        accounts
```

Comment:
It could be useful to extract those 3 lines in a function with a meaningful name, so that we don't need a comment to explicit the behavior and there is no future risk of having both functions diverge.

Reply:
Fair point! Will do.